File: deps/wazuh_testing/wazuh_testing/tools/macos_log/log_generator.m
#import <os/activity.h>
#import <os/log.h>
#import <os/trace.h>

#define custom_os_log_message "Custom os_log event message"

int main(int argc, char ** argv){
    char *type = argv[1];
    char *level = argv[2];
    char *subsystem = argv[3];
    char *category = argv[4];

    if(strcmp(type, "activity") == 0){
        os_activity_initiate("Custom activity event message", OS_ACTIVITY_FLAG_DEFAULT, ^{});
    }
    else if(strcmp(type, "trace") == 0){
        os_trace("Custom trace event message");
    }
    else{
        os_log_t log = os_log_create(subsystem, category);
        if(strcmp(level, "info") == 0){
            os_log_info(log, custom_os_log_message);
        }
        else if(strcmp(level, "debug") == 0){
            os_log_debug(log, custom_os_log_message);
        }
        else if(strcmp(level, "error") == 0){
            os_log_error(log, custom_os_log_message);
        }
        else if(strcmp(level, "fault") == 0){
            os_log_fault(log, custom_os_log_message);
        }
        else{
            os_log(log, custom_os_log_message);
        }
    }
}

File: deps/wazuh_testing/wazuh_testing/tools/macos_log/log_generator.swift
import os.log

let log_parameters = CommandLine.arguments
let type_event = log_parameters[1]
let oslog_subsystem_category = OSLog(subsystem: log_parameters[2], category: log_parameters[3])

var oslog_type = OSLogType.default

switch type_event{
    case "info":
        oslog_type = OSLogType.info
    case "debug":
        oslog_type = OSLogType.debug
    case "default":
        oslog_type = OSLogType.default
    case "error":
        oslog_type = OSLogType.error
    case "fault":
        oslog_type = OSLogType.fault
    default:
        oslog_type = OSLogType.default
}

os_log("Custom os_log event message", log: oslog_subsystem_category, type: oslog_type)
